### ISSUE TITLE
Suggestion: Is it possible to apply Reduce to Pole transformation using per pixel D and I?

### ISSUE THREAD

**Issue by brunorpinho** [fatiando/fatiando#400]
Hello,

I am working in a huge area with magnetic data and I used fatiando's RTP implementation to find the "vertical" magnetic anomalies, the results were great. However, due to variations in declination and inclination of the IGRF12 model across the area (see image) I fell that my results could be improved by passing 1D arrays of "D"s and "I"s for each coordinate (like x, y and data were passed). 

Is it possible? (I am not very familiar with the frequency domain.)

![image](https://cloud.githubusercontent.com/assets/11368602/26513370/7cca9c00-4241-11e7-9536-82ced8a47d10.png)

**Comment by leouieda** [comment]
Hi @brunorpinho, why did you close the issue? Did you get this resolved?

In the frequency domain, you really can't do this. The assumption is that you have sources with a single magnetization direction in a uniform inducing field. 

In fact, the problem isn't just the IGRF variation. In such a huge area, you'll surely have sources with different magnetizations. That's why [our implementation](http://www.fatiando.org/dev/api/gravmag.transform.html#fatiando.gravmag.transform.reduce_to_pole) requires you to input the source magnetization direction explicitly. Many commercial software don't ask for this and implicitly assume induced magnetization only. That can be **very** misleading.

My general advice is always to **never RTP large surveys with multiple sources**. RTP is great for single sources (or multiple sources with the same magnetization, like a dike swarm). You can use it as a diagnosis tool to check for remanent magnetization. But it's not a "processing" transformation to enhance anomalies. Use the Total Gradient Amplitude instead.

**Comment by brunorpinho** [comment]
Hello @leouieda !

Thanks for this explanations, this is what I needed some months ago. :(

I am studying the application of machine learning for geological mapping using aeromagnetic and aerogamma-ray data. So I need a way to reduce the high dipolar behavior in the study area. This way the Magnetic data will "correlate" (or not) better with the Gamma-ray data.

After some analysis I put down the RTP, because I figured out during the processing that the magnetic anomalies lost their structural aspects - mainly beautiful folds, faults and dykes related to a strike-slip transpressional system and a bunch of other interpretable structures. This is totally related to what you are explaining to me right now, I have multiple sources of different magnetization.

I ended up using the remanent magnetization and its 1st vertical derivative for my preliminary results, as shown below.

![15_mag_rtp_1dv](https://user-images.githubusercontent.com/11368602/28802982-d8832ea0-762f-11e7-8c52-ba6196f8df8c.png)

If I continue my research I will try the Total Gradient Amplitude for sure.

Thanks Leo!
